--- origsrc/xmms2-0.6DrMattDestruction/src/clients/lib/perl/wscript	2009-04-21 12:51:11.000000000 -0500
+++ src/xmms2-0.6DrMattDestruction/src/clients/lib/perl/wscript	2009-07-20 20:37:49.888269500 -0500
@@ -17,9 +17,9 @@ def build(bld):
     obj.mac_bundle = True
 
     bld.install_files('${INSTALLDIR_PERL_LIB}', 'pm/XMMSClient.pm')
-    bld.install_files('${INSTALLDIR_PERL_LIB}', 'pm/XMMSClient/Sync.pm')
-    bld.install_files('${INSTALLDIR_PERL_LIB}', 'pm/XMMSClient/Glib.pm')
-    bld.install_files('${INSTALLDIR_PERL_LIB}', 'pm/XMMSClient/Collection.pm')
+    bld.install_files('${INSTALLDIR_PERL_LIB}/XMMSClient', 'pm/XMMSClient/Sync.pm')
+    bld.install_files('${INSTALLDIR_PERL_LIB}/XMMSClient', 'pm/XMMSClient/Glib.pm')
+    bld.install_files('${INSTALLDIR_PERL_LIB}/XMMSClient', 'pm/XMMSClient/Collection.pm')
 
 def configure(conf):
     conf.check_tool('perl')
