--- origsrc/xmms2-0.7DrNo/src/clients/lib/perl/wscript	2010-02-19 11:43:54.000000000 -0600
+++ src/xmms2-0.7DrNo/src/clients/lib/perl/wscript	2010-05-24 10:51:54.601501100 -0500
@@ -24,8 +24,7 @@ def build(bld):
 def configure(conf):
     conf.check_tool('perl')
 
-    if not conf.check_perl_version((5,7,3)):
-        conf.fatal("Wrong perl version")
+    conf.check_perl_version((5,7,3))
 
     conf.check_perl_ext_devel()
 
