--- origsrc/xmms2-0.8DrO_o/.waf-1.6.7-b271784c448ab906d5606267fb7f92c0/waflib/Tools/perl.py	2011-12-11 16:33:17.274298100 -0600
+++ src/xmms2-0.8DrO_o/.waf-1.6.7-b271784c448ab906d5606267fb7f92c0/waflib/Tools/perl.py	2011-12-11 16:35:14.180902000 -0600
@@ -57,10 +57,10 @@ def check_perl_ext_devel(self):
 		self.fatal('find perl first')
 	def read_out(cmd):
 		return Utils.to_list(self.cmd_and_log(perl+cmd))
-	env['LINKFLAGS_PERLEXT']=read_out(" -MConfig -e'print $Config{lddlflags}'")
+	env['LINKFLAGS_PERLEXT']=read_out(" -MConfig -e'print \"$Config{lddlflags} -L$Config{archlibexp}/CORE -lperl\"'")
 	env['INCLUDES_PERLEXT']=read_out(" -MConfig -e'print \"$Config{archlib}/CORE\"'")
 	env['CFLAGS_PERLEXT']=read_out(" -MConfig -e'print \"$Config{ccflags} $Config{cccdlflags}\"'")
-	env['XSUBPP']=read_out(" -MConfig -e'print \"$Config{privlib}/ExtUtils/xsubpp$Config{exe_ext}\"'")
+	env['XSUBPP']=read_out(" -MConfig -e'print \"$Config{privlib}/ExtUtils/xsubpp\"'")
 	env['EXTUTILS_TYPEMAP']=read_out(" -MConfig -e'print \"$Config{privlib}/ExtUtils/typemap\"'")
 	if not getattr(Options.options,'perlarchdir',None):
 		env['ARCHDIR_PERL']=self.cmd_and_log(perl+" -MConfig -e'print $Config{sitearch}'")
